statsmodels/tools/data_interface.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import numpy as np
+from functools import partial
+from patsy import dmatrix
+
+NUMPY_TYPES = [np.ndarray, np.float64]
+
+class DataInterface(object):
+
+    def __init__(self, permitted_types, internal_type, data=None, external_type=None, model=None, use_formula=False):
```

Comment:
I think `internal_type` should default to `np.ndarray` which is currently almost always the case